src/util/std_types.h

```diff
@@ -636,34 +576,27 @@ class enumeration_typet:public typet
   }
 };
 
-/*! \brief Cast a generic typet to a \ref enumeration_typet
- *
- * This is an unchecked conversion. \a type must be known to be \ref
- * enumeration_typet.
- *
- * \param type Source type
- * \return Object of type \ref enumeration_typet
- *
- * \ingroup gr_std_types
-*/
+/// \brief Cast a generic typet to a \ref enumeration_typet
+///
+/// This is an unchecked conversion. \param type must be known to be \ref
+/// enumeration_typet.
+///
+/// \param type Source type.
+/// \return Object of type \ref enumeration_typet.
 inline const enumeration_typet &to_enumeration_type(const typet &type)
 {
   PRECONDITION(type.id()==ID_enumeration);
   return static_cast<const enumeration_typet &>(type);
 }
 
-/*! \copydoc to_enumeration_type(const typet &)
- * \ingroup gr_std_types
-*/
+/// \copydoc to_enumeration_type(const typet &)
 inline enumeration_typet &to_enumeration_type(typet &type)
 {
   PRECONDITION(type.id()==ID_enumeration);
   return static_cast<enumeration_typet &>(type);
 }
 
-/*! \brief The type of C enums
-*/
-
+/// The type of C enums
```

Comment:
Do Java and C++ enums derive from this?

Reply:
No, there's no class extending this. Seems to only represent C enums indeed